src/counterexamples/canonically_ordered_comm_semiring_two_mul.lean

```diff
@@ -0,0 +1,289 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.zmod.basic
+import ring_theory.subsemiring
+import algebra.ordered_monoid
+/-!
+
+A `canonically_ordered_comm_semiring` with two different elements `a` and `b` such that
+`a ≠ b` and `2 * a = 2 * b`.  Thus, multiplication by a fixed non-zero element of a canonically
+ordered semiring need not be injective.  In particular, multiplying by a strictly positive element
+need not be strictly monotone.
+
+Recall that a `canonically_ordered_comm_semiring` is a commutative semiring with a partial ordering
+that is "canonical" in the sense that the inequality `a ≤ b` holds if and only if there is a `c`
+such that `a + c = b`.  There are several compatibility conditions among addition/multiplication
+and the order relation.  The point of the counterexample is to show that monotonicity of
+multiplication cannot be strengthened to **strict** monotonicity.
+
+Reference:
+https://
+leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/canonically_ordered.20pathology
+-/
+
+namespace from_Bhavik
+
+/--  Bhavik Mehta's example. -/
+@[derive [comm_semiring]]
+def K : Type := subsemiring.closure ({1.5} : set ℚ)
```

Comment:
This example is given here, but almost nothing is proven about it and (unless I'm missing something) it is not used in the example below. Can you explain in (module) doc strings why this example is here and what it is a counterexample of?

Reply:
You are right: this is an example showing that the inequality `2 * a \le 2 * b` need not imply the inequality `a \le b`, even with lots of canonically ordered properties.  Bhavik has a proof, but I have not chased it down.  I still wanted to record it, since it is a nice example to keep in mind!